File: schemas/multi-language-i18n/drizzle/machine_translation_configs.ts
// machine_translation_configs: Configuration for machine translation engines.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, integer, jsonb, numeric, timestamp } from "drizzle-orm/pg-core";

export const machineTranslationConfigs = pgTable(
  "machine_translation_configs",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    engine: text("engine").notNull(),
    isEnabled: boolean("is_enabled").notNull().default(true),
    isDefault: boolean("is_default").notNull().default(false),
    apiKeyRef: text("api_key_ref"),
    endpointUrl: text("endpoint_url"),
    supportedLocales: jsonb("supported_locales"),
    defaultQualityScore: numeric("default_quality_score", {
      precision: 3,
      scale: 2,
    }),
    rateLimit: integer("rate_limit"),
    options: jsonb("options"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true })
      .notNull()
      .defaultNow()
      .$onUpdateFn(() => new Date()),
  }
);

File: schemas/multi-language-i18n/spacetimedb/machine_translation_configs.rs
// machine_translation_configs: Configuration for machine translation engine integrations.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = machine_translation_configs, public)]
pub struct MachineTranslationConfig {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,
    pub engine: String,
    pub is_enabled: bool,
    pub is_default: bool,
    pub api_key_ref: Option<String>,
    pub endpoint_url: Option<String>,
    pub supported_locales: Option<String>, // JSON
    pub default_quality_score: Option<f64>,
    pub rate_limit: Option<i32>,
    pub options: Option<String>, // JSON
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
